src/defaults/settings_dsl.rs

```diff
@@ -1,119 +1,222 @@
+        )
+    };
+    ($id:ident, =int $first:literal ..= $last: literal $(@ $radix: literal)?; $value: expr) => {
+        #[allow(clippy::from_str_radix_10)]
+        $crate::defaults::SettingKind::Integer(
+            |text| i64::from_str_radix(text, 10$(*0 + $radix)?)
+                            .ok()
+                            .filter(|val| ($first ..= $last)
+                            .contains(val))
+                            .map(|i| Box::new(move |obj: &mut Settings| obj.$id = i) as Box<dyn FnOnce(&mut Settings)>)
+        )
+    };
+    ($id:ident, =int $fn: expr; $value: expr) => {
+        $crate::defaults::SettingKind::Integer(
+            |text| $fn(&text).map(|i| Box::new(move |obj: &mut Settings| obj.$id = i) as Box<dyn FnOnce(&mut Settings)>)
+        )
+    };
+    ($id:ident, $(=int $check: expr;)+ $value: expr) => { compile_error!("bla") };
+    ($id:ident, $(=enum $key: ident;)+ $value: ident) => {
+        $crate::defaults::SettingKind::Text(|key| match key {
+            $(
+            stringify!($key) => { Some(Box::new(move |obj: &mut Settings| obj.$id = $crate::defaults::enums::$id::$key)) },
+            )*
+            _ => None,
+        })
+    };
+    ($id:ident, None) => {
+        $crate::defaults::SettingKind::Text(
+            |text| {
+                let text = text.into();
+                Some(Box::new(move |obj: &mut Settings| obj.$id = Some(text)))
             }
-        }
-
-        impl ToSudoDefault for ($type, $type) {
-            #[allow(clippy::from_str_radix_10)]
-            fn to_sudodefault(self) -> SudoDefault {
-                let (value, neg) = self;
-                $crate::defaults::SudoDefault::$ctor(OptTuple {
-                    default: value.into(),
-                    negated: Some(neg.into()),
-                }$(, $vetting_function)?)
+        )
+    };
+    ($id:ident, $value: expr) => {
+        $crate::defaults::SettingKind::Text(
+            |text| {
+                let text = text.into();
+                Some(Box::new(move |obj: &mut Settings| obj.$id = Some(text)))
             }
-        }
+        )
     };
 }
 
-macro_rules! sliceify {
-    ([$($value:tt),*]) => {
-        &[$($value),*][..]
+macro_rules! has_standard_negator {
+    (true) => {
+        true
+    };
+    (false) => {
+        true
+    };
+    ([$($_: expr),*]) => {
+        true
     };
-    ($value:tt) => {
-        ($value)
+    ($($_: tt)*) => {
+        false
     };
 }
 
-macro_rules! tupleify {
-    ($fst:expr, $snd:expr) => {
-        ($fst, $snd)
+// this macro allows us to help the compiler generate more efficient code in 'fn negate'
+// and enables the way 'fn set' is made
+macro_rules! ifdef {
+    (; $then: expr; $else: expr) => {
+        $else
     };
-    ($value:tt) => {
-        $value
+    ($($_: expr)+; $then: expr; $else: expr) => {
+        $then
     };
-}
 
-macro_rules! optional {
     () => {
-        |x| x
+        return None
     };
```

Comment:
Looks like this branch is only hit by `initializer_of!()` which can use `return None` directly.

Reply:
Correct